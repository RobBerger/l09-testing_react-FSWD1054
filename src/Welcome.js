import UserName from './UserName'

function Welcome() {
    function currentDate() {
        return new Date(Date.now()).toLocaleString()
    }
  return (
    <>
    <h1 role="heading">Welcome back, <UserName /></h1>
    <div data-testid="footer">This page was rendered on {currentDate()}</div>
    </>
  )
}
export default Welcome