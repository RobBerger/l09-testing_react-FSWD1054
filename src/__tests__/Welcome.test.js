import React from 'react';
import { render, screen } from '@testing-library/react';
import Welcome from '../Welcome';

jest.mock("../UserName", () => function UserName() {
    return (<span>Test User</span>)
})

it('displays the user name in the heading', () => {
  render(<Welcome />)
  expect(screen.getByRole('heading', {name: 'Welcome back, Test User'})).toBeInTheDocument();
})

afterEach(() => {
    jest.restoreAllMocks()
  })

  describe('the welcome footer', () => {
    it('retrieves the current date', () => {
      const spy = jest.spyOn(Date, 'now')
      render(<Welcome />)
      expect(spy).toHaveBeenCalled()
    })
  
    it('displays the current date', () => {
      const spy = jest.spyOn(Date, 'now')
        .mockImplementation(() => new Date('2021-01-01T00:00')
      )
      render(<Welcome />)
      expect(screen.getByText('This page was rendered on 1/1/2021, 12:00:00 AM')).toBeInTheDocument()
    })
  })