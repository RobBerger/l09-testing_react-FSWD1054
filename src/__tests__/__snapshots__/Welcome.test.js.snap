// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <h1
    role="heading"
  >
    Welcome back, 
    <span>
      Test User
    </span>
  </h1>
  <div
    data-testid="footer"
  >
    This page was rendered on 1/1/2021, 12:00:00â€¯AM
  </div>
</DocumentFragment>
`;
